sql-plugin/src/main/311until330-all/scala/com/nvidia/spark/rapids/shims/GpuTypeShims.scala

```diff
@@ -82,15 +83,55 @@ object GpuTypeShims {
     throw new RuntimeException(s"Not support type $dt.")
 
   /**
-   * Spark supports interval type from 320; Spark supports to/from Parquet from 330,
-   * And lots of interval operators from 330, so just return false for Spark330-
+   * Alias supported types for this shim
    */
-  def isDayTimeIntervalType(dt: DataType) : Boolean = false
+  def supportedTypesForAlias: TypeSig = TypeSig.none
 
   /**
-   * Spark supports interval type from 320; Spark supports to/from Parquet from 330,
-   * And lots of interval operators from 330, so just return false for Spark330-
+   * Abs supported types for this shim
    */
-  def isYearMonthIntervalType(dt: DataType) : Boolean = false
+  def supportedTypesForAbs: TypeSig = TypeSig.none
```

Comment:
My original comment was about the name being too generic, but now it looks like the names are too specific.  Having to add three new methods to the type shims for every new expression that might touch them is a lot of overhead.  Do we really need this granularity?